packages/rolldown/src/plugin/hook-filter.ts

```diff
@@ -1,18 +1,19 @@
 import type { StringOrRegExp } from '../constants/types'
 import type { ModuleType } from '../index'
 
-export type BaseHookFilter = {
-  id?: {
-    include?: StringOrRegExp[]
-    exclude?: StringOrRegExp[]
-  }
-  moduleType?:
-    | ModuleType[]
-    | {
-        include?: ModuleType[]
-      }
-  code?: {
-    include?: StringOrRegExp[]
-    exclude?: StringOrRegExp[]
-  }
+interface StringMatcher {
```

Comment:
`StringMatcher` is named due to the matching target is `string`. I don't want call it `filter` because we already has a `HookFilter` below. They are not in the same concept level, so I use `matcher` to as the difference.